scoap3/modules/compliance/compliance.py

```diff
@@ -0,0 +1,226 @@
+            if isinstance(m, tuple):
+                matches.append(m[0])
+            else:
+                matches.append(m)
+
+    return matches
+
+
+def __find_regexp_in_pdf(obj, patterns, forbidden_patterns=None):
+    """
+    Finds all matches for given patterns with surrounding characters in all filetypes.
+    Fails only if there are no matches at all or there is a match for a forbidden pattern.
+    :param patterns: iterable of string patterns
+    """
+    __extract_text_as_extra_data(obj)
+
+    ok = True
+    details = []
+
+    for filetype, data in obj.extra_data['extracted_data'].iteritems():
+        # first check for forbidden patterns
+        if forbidden_patterns:
+            forbidden_matches = __find_regexp(data, forbidden_patterns)
+            if forbidden_matches:
+                ok = False
+                details.append('Found forbidden match in %s: "%s"' % (filetype, '", "'.join(set(forbidden_matches))))
+
+        matches = __find_regexp(data, patterns)
+        if not matches:
+            ok = False
+            details.append('Not found in %s' % filetype)
+        else:
+            details.append('Found in %s as: "%s"' % (filetype, '", "'.join(set(matches))))
+
+    return ok, details, None
+
+
+def _files(obj):
+    """check if it has the necessary files: .xml, .pdf, .pdfa """
+
+    file_types = [file['filetype'] for file in obj.data['_files']]
+
+    ok = True
+    details = ''
+
+    if 'xml' not in file_types:
+        ok = False
+        details += 'No xml file. '
+
+    if 'pdf' not in file_types and 'pdf/a' not in file_types:
+        ok = False
+        details += 'No pdf file. '
+
+    details += 'Available files: %s' % ', '.join(file_types)
+
+    return ok, (details, ), None
+
+
+def _received_in_time(obj):
+    """check if publication is not older than 24h """
+    api_url = 'https://api.crossref.org/works/%s'
+    api_message = requests.get(api_url % __get_doi(obj)).json()['message']
+
+    if 'publication_info' in obj.data and obj.data['publication_info'][0]['journal_title'] == 'Progress of Theoretical and Experimental Physics':
+        parts = api_message['published-online']['date-parts'][0]
+        # only contains day of publication, check for end of day
+        api_time = datetime(*parts, hour=23, minute=59, second=59)
+    else:
+        api_time = parse_date(api_message['created']['date-time'], ignoretz=True)
+    received_time = parse_date(obj.data['acquisition_source']['date'])
+    delta = received_time - api_time
+
+    ok = delta <= timedelta(hours=24)
+    details_message = 'Arrived %d hours later then creation date on crossref.org.' % (delta.total_seconds() / 3600)
+    debug = 'Time from crossref: %s, Received time: %s' % (api_time, received_time)
+
+    return ok, (details_message ,), debug
+
+
+def _founded_by(obj):
+    """check if publication has "Founded by SCOAP3" marking *in pdf(a) file* """
+
+    patterns = ['scoap3?', ]
```

Comment:
Is there any reason why we should not check for the full phrase "funded by scoap3", is it just simplification?